pkg/kube/pod_controller.go

```diff
@@ -211,26 +211,20 @@ func (p *podController) StopPod(ctx context.Context, stopTimeout time.Duration,
 	return nil
 }
 
-// getContainerName returns container name, which should be passed to
-// operations that require it.
-// If the container name was specified in podOptions, it will be used.
-// Otherwise, the first container name from specs will be taken as best effort
-// (when pods are created with sidecars, sidecar containers are placed after
-// main container).
-func (p *podController) getContainerName() string {
-	if p.podOptions.ContainerName != "" {
-		return p.podOptions.ContainerName
+func ContainerNameFromPodOpts(po *PodOptions) string {
```

Comment:
If this is not limited to `podController`, we should move it to `pod.go`.
Additionally, I preferred the previous name. It was generic. It's getting the name from `PodOptions` only when available. So, `GetContainerName` or `ContainerNameFromPodOptsOrDefault` would be more appropriate.
The comment section should be modified and added back as well.